spec/integration/startup-spec.coffee

```diff
@@ -227,3 +227,13 @@ describe "Starting Atom", ->
                 [tempDirPath]
                 [otherTempDirPath]
               ].sort()
+
+  describe "opening a remote directory", ->
+    it "opens the parent directory and creates an empty text editor", ->
+      remoteDirectory = 'remote://server:3437/some/directory/path'
+      runAtom [remoteDirectory], {ATOM_HOME: atomHome}, (client) ->
+        client
+          .waitForWindowCount(1, 1000)
+          .waitForExist("atom-workspace", 5000)
+          .treeViewRootDirectories()
+          .then ({value}) -> expect(value).toEqual([remoteDirectory])
```

Comment:
That weird state was already the case, and it was handled by us later when our package is activated to replace the root directory with its remote version.
All what this PR guarantees is that the path survives normalization, but doesn't fix the fact that Atom core initializes and deserializes state before the third party packages get a say in how that should happen.

However, I'd be happy to help in a following PR that does something like what @bolinfest suggested before:

**Either:**
When serializing the project state, include which directory provider (by names) created this directory and when reloading the state, wait for that specific directory provider to register and create the directories at that time

**OR:**
Add a function to directory providers:
```js
handlePath(path: string): boolean
```
where that would be in `default-directory-provider.coffee`:
```coffee
handlePath(directoryPath) ->
  fs.existsSync(directoryPath) or fs.existsSync(path.dirname(directoryPath))
```
and in remote directory provider:
```coffee
handlePath(directoryPath) ->
  directoryPath.startsWith('remote://')
```